### PR TITLE
Nuke: Render Workfile Attributes

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
`Workfile Dependency` default value can now be controlled from project settings.
`Use Published Workfile` makes using published workfiles for rendering optional.

## Testing notes:
1. Toggle `deadline/publish/NukeSubmitDeadline/workfile_dependency` off.
2. Create Write instance in Nuke and check the `Workfile Dependency` is off.
3. Toggle `Use Published Workfile` and validate the Deadline job is using the work area workfile.

### CODE REVIEW

**Comment by tokejepsen** [comment]
Docs https://github.com/ynput/ayon-documentation/pull/126

**Comment by jakubjezek001** [comment]
looks good, we will also need to bump the version in `server_addon\nuke\server\version.py`

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
OpenPype settings works as expected but Ayon server is not workfing well. Please check default settings if there are correctly set. 

```
Traceback (most recent call last):
2024-01-18 14:57:56       File "/backend/api/settings/settings.py", line 178, in get_all_settings
2024-01-18 14:57:56         settings = await addon.get_studio_settings(variant)
2024-01-18 14:57:56                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-01-18 14:57:56       File "/backend/ayon_server/addons/addon.py", line 288, in get_studio_settings
2024-01-18 14:57:56         settings = await self.get_default_settings()
2024-01-18 14:57:56                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-01-18 14:57:56       File "/addons/deadline/0.1.7/__init__.py", line 17, in get_default_settings
2024-01-18 14:57:56         return settings_model_cls(**DEFAULT_VALUES)
2024-01-18 14:57:56                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-01-18 14:57:56       File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
2024-01-18 14:57:56     pydantic.error_wrappers.ValidationError: 1 validation error for DeadlineSettings
2024-01-18 14:57:56     publish -> NukeSubmitDeadline -> use_published_workfile
2024-01-18 14:57:56       field required (type=value_error.missing)
2024-01-18 14:57:56     
```

**Comment by jakubjezek001** [comment]
> > looks good, we will also need to bump the version in server_addon\nuke\server\version.py
> 
> Good point! Done.

Sorry for my mistake in path but it should be increased in deadline server addon not nuke.

File: server_addon/deadline/server/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.1.7"
+__version__ = "0.2.7"
```
Comment (jakubjezek001) [inline server_addon/deadline/server/version.py:1]:
```suggestion
__version__ = "0.1.8"
```

**Review by jakubjezek001 — APPROVED** [review]
LGTM!

File: openpype/modules/deadline/plugins/publish/submit_nuke_deadline.py [hunk 2]
```diff
@@ -125,20 +131,27 @@ def process(self, instance):
...
-                script_path = os.path.normpath(template_filled)
-
-                self.log.info(
-                    "Using published scene for render {}".format(script_path)
-                )
+        use_published_workfile = instance.data["attributeValues"].get(
+            "use_published_workfile", self.use_published_workfile
+        )
+        if use_published_workfile:
+            for item_ in context:
```
[the remark this answers is not in the document]
Comment (tokejepsen) [reply openpype/modules/deadline/plugins/publish/submit_nuke_deadline.py:138]: Think the issue is actually that the plugin is not inherited from; https://github.com/ynput/OpenPype/blob/develop/openpype/modules/deadline/abstract_submit_deadline.py#L412, which already has the "Use Published Workfile" logic.
Think inheriting from `AbstractSubmitDeadline`, might too big of a job for this PR, so maybe we should just note it down as an issue?

[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/modules/deadline/plugins/publish/submit_nuke_deadline.py:138]: This suggestion is out of scope of this PR, so this comment can be resolved.

But it still deservers a function and should be added more checks (and comments).
```python
    def _get_published_workfile_path(self, context):
        for instance in context:
            if (
                instance.data["family"] != "workfile"
                # Disabled instances won't be integrated
                or instance.data("publish") is False
            ):
                continue
            template_data = instance.data["anatomyData"]
            # Expect workfile instance has only one representation
            representation = instance.data["representations"][0]
            # Get workfile extension
            repre_file = representation["files"][0]
            ext = os.path.splitext(repre_file)[1].lstrip(".")

            # Fill template data
            template_data["representation"] = representation["name"]
            template_data["ext"] = ext
            template_data["comment"] = None

            anatomy = context.data["anatomy"]
            # NOTE: Hardcoded template name 'publish' > may not be used
            template_obj = anatomy.templates_obj["publish"]["path"]

            template_filled = template_obj.format(template_data)
            script_path = os.path.normpath(template_filled)
            self.log.info(
                "Using published scene for render {}".format(
                    script_path
                )
            )
            return script_path
        return None
```

[the remark this answers is not in the document]
Comment (tokejepsen) [reply openpype/modules/deadline/plugins/publish/submit_nuke_deadline.py]: Cool, have implemented and added method description so we know for later.

**Pull request merged**